pi_ldapproxy/util.py

```diff
@@ -1,9 +1,25 @@
+from OpenSSL.SSL import SSL_CB_HANDSHAKE_DONE
 from twisted.internet._sslverify import OpenSSLCertificateOptions, ClientTLSOptions
+from twisted.internet.interfaces import IOpenSSLClientConnectionCreator
 from twisted.web.client import _requireSSL
 from twisted.web.iweb import IPolicyForHTTPS
 from zope.interface import implementer
 
 
+@implementer(IOpenSSLClientConnectionCreator)
+class DisabledVerificationClientTLSOptions(ClientTLSOptions):
+    def _identityVerifyingInfoCallback(self, connection, where, ret):
+        """
+        Do not check the remote hostname
+        """
+        if where & SSL_CB_HANDSHAKE_DONE:
```

Comment:
To my understanding this if-branch does not check the hostname?

So the comment "Do not check the hostname" should be after the if and not before the if, right?

Reply:
In that sense, yes. The method we are overriding here, ``ClientTLSOptions._identityVerifyingInfoCallback``, performs hostname validation if ``where & SSL_CB_HANDSHAKE_DONE``. So a better docstring would be "Variant of ``_identityVerifyingInfoCallback`` which does not perform hostname validation".